### PR TITLE
Calendar Intent

### PR DESCRIPTION
**Pull request by Kuriv2001**
This PR adds the ability to create calendar events from chimpagne events.

- A function has been added that allows calendar intents to be created from a chimpagne event.
- A pop up has been added to ask if the user want to add this event to his calendar. If the user answers yes the intent will be launched, on dismiss/no nothing will happen.
- A button has been added, that calls the pop up when pressed => integrated in viewDetailedEventsScreen, not final place yet (will be changed by @leagrieder)
- Implemented when joining an event from the FindEventsScreen

Created intent for Calendar app has the: title, start date/time and end date/time, location of the event + a reminder/alert for a day before the event.

Use from viewDetailedEventPage:
Screenshot 2024-05-12 at 14 51 11
If yes is then pressed:
Screenshot 2024-05-12 at 14 51 20

Use from DiscoverEventsScreen:
Screenshot 2024-05-12 at 14 54 48

### CODE REVIEW

File: app/src/main/java/com/monkeyteam/chimpagne/ui/DetailScreenSheet.kt [hunk 1]
```diff
@@ -52,11 +69,22 @@ fun DetailScreenSheet(event: ChimpagneEvent?, onJoinClick: () -> Unit = {}) {
...
+              onClick = enhancedOnJoinClick,
               modifier = Modifier.align(Alignment.CenterHorizontally).testTag("join_button")) {
                 Text(stringResource(id = R.string.find_event_join_event_button_text))
               }
         }
+    if (showDialog && context != null) {
+      popUpCalendar(
+          onAccept = {
+            createCalendarIntent(event)?.let { context.startActivity(it) }
+                ?: Toast.makeText(context, "Event can't be added to calendar", Toast.LENGTH_SHORT)
```
Comment (leagrieder) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/DetailScreenSheet.kt:81]: Great Work! Maybe this Toast can be made as a StringResource ?

File: app/src/main/java/com/monkeyteam/chimpagne/ui/components/Button.kt [hunk 2]
```diff
@@ -97,3 +108,50 @@ fun GoBackButton(navigationActions: NavigationActions) {
...
+  }
+}
+
+@Composable
+fun popUpCalendar(onAccept: () -> Unit, onReject: () -> Unit, event: ChimpagneEvent?) {
+  val builder =
+      AlertDialog(
+          onDismissRequest = { onReject() },
+          title = { Text("Add to Calendar") },
+          text = { Text("Do you want to add the event \"${event?.title}\" to your calendar?") },
```
Comment (leagrieder) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/components/Button.kt:146]: Here as well, make it as a StringResource

**Review by leagrieder — CHANGES_REQUESTED** [review]
Great job on adding the calendar events! Really cool feature that you implemented gracefully.  The modifications made to the codebase are consistent with the existing style and patterns. The implementation of user feedback using `Toast` messages is good practice and just need to be made as StringResources, everything else looks great.

This pull request is well-crafted with clear intent and good implementation quality.

File: app/src/main/java/com/monkeyteam/chimpagne/ui/DetailScreenSheet.kt (hunk 1 above)
Comment (Kuriv2001) [reply app/src/main/java/com/monkeyteam/chimpagne/ui/DetailScreenSheet.kt]: done

File: app/src/main/java/com/monkeyteam/chimpagne/ui/components/Button.kt (hunk 2 above)
Comment (Kuriv2001) [reply app/src/main/java/com/monkeyteam/chimpagne/ui/components/Button.kt]: done

**Pull request merged**